### ISSUE TITLE
Classification

### ISSUE THREAD

**Issue by akald** [amaiya/ktrain#251]
I'm interested in understanding how the classifier works. Is classification based on the maximum entropy method, or what's the process for which the label and the probabilities (get_proba) are generated like in the IMDB reviews example.

I couldn't seem to find an answer in the documentation, sorry.

Many thanks in advance!

**Comment by amaiya** [comment]
This completely depends on which model is selected.  If you type `text.print_text_classifiers()`, there are links to research papers describing the details of each model supported by the `text_classifier` API:
```python
from ktrain import text
text.print_text_classifiers()     
# OUTPUT:                         
# fasttext: a fastText-like model [http://arxiv.org/pdf/1607.01759.pdf]
# logreg: logistic regression using a trainable Embedding layer
# nbsvm: NBSVM model [http://www.aclweb.org/anthology/P12-2018]
# bigru: Bidirectional GRU with pretrained fasttext word vectors [https://fasttext.cc/docs/en/crawl-vectors.html]
# standard_gru: simple 2-layer GRU with randomly initialized embeddings
# bert: Bidirectional Encoder Representations from Transformers (BERT) [https://arxiv.org/abs/1810.04805]
# distilbert: distilled, smaller, and faster BERT from Hugging Face [https://arxiv.org/abs/1910.01108]
```

For instance, if selecting `bert`, then you can read [this paper](https://arxiv.org/abs/1810.04805).   You can also just look at the source code to see how the model is constructed.  This is useful for models like `nbsvm`, `fasttext`, `bigru`, etc.

Also, when you use the `return_proba=True` argument, it simply returns the raw `softmax` output of the model.  When `return_proba=False`, the label with the maximum `softmax` probability is displayed.`

See the [softmax Wikipedia page](https://en.wikipedia.org/wiki/Softmax_function) for more information on `softmax`.

**Comment by akald** [comment]
Thanks, sorry last question I'm getting confused by the terminology.
For Bert, with two labels (positive or negative sentiment) is the SoftMax/cross-entropy loss method the same as maximum entropy?

**Comment by amaiya** [comment]
Maximum entropy classification (i.e., MaxEnt) is just a fancy word for logistic regression (the `logreg` model in ktrain).  See Slide 23 in [this PDF](https://web.stanford.edu/class/cs124/lec/Maximum_Entropy_Classifiers.pdf), for example. It has nothing to do with BERT really.
